app/models/ht_user.rb

```diff
@@ -50,7 +50,30 @@ def expires(short: false)
     short ? self[:expires]&.strftime('%Y-%m-%d') : self[:expires]&.to_s(:db)
   end
 
+  # How many days until expiration?
+  # @return [Number] days until expiration
+  def days_until_expiration
+    seconds_until_expiration = self[:expires] - Time.zone.now
```

Comment:
Two issues:

1) In development at least, I get a `String` for self[:expires], not a `Time` (despite the column being a datetime in sqlite3...); haven't checked with mysql in production.
2) If one does get a Time, is there a reason not to use `self[:expires].to_date - Date.today`, and omit the `days_from_seconds` helper method? Alternatively, `Date.parse(self[:expires]) - Date.today`, if you actually get a string even in production...

Reply:
1. It's actually a rails ActiveSupport::TimeWithZone object. They've change `#inspect`, `#pretty_print`, `#to_s`, etc. to always show the string representation, but it's a real timzone object underneath.

2. For two dates, d1 - d2 gives a Rational, but I can obviously cast that to an integer. I'll do that, and the other suggested changes on this PR.

Reply:
Ah, thanks... I was looking with `pry`, but didn't look close enough...